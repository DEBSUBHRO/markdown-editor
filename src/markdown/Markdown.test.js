import PluginManager from '../PluginManager';
import FromMarkdown from './fromMarkdown';
import ToMarkdown from './toMarkdown';
import List from '../plugins/list';

let fromMarkdown = null;
let toMarkdown = null;

beforeAll(() => {
  const plugins = [List()];
  const pluginManager = new PluginManager(plugins);
  fromMarkdown = new FromMarkdown(pluginManager);
  toMarkdown = new ToMarkdown(pluginManager);
});

/* Roundtrips */

test('can roundtrip basic text', () => {
  const markdownText = 'This is some text.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip italic text', () => {
  const markdownText = 'This is *some* text.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip bold text', () => {
  const markdownText = 'This is **some** text.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip bold italic text', () => {
  const markdownText = 'This is ***some*** text.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h1 to slate', () => {
  const markdownText = '# Heading One';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h2', () => {
  const markdownText = '## Heading Two';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h3', () => {
  const markdownText = '### Heading Three';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h4', () => {
  const markdownText = '#### Heading Four';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h5', () => {
  const markdownText = '##### Heading Five';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip h6', () => {
  const markdownText = '###### Heading Six';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip paragraphs', () => {
  const markdownText = `This is first paragraph.
  
  This is second paragraph.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip inline code', () => {
  const markdownText = 'This is `inline code`.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a link', () => {
  const markdownText = 'This is [a link](http://clause.io).';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a thematic break', () => {
  const markdownText = `This is 
  
  ---
  A footer.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a blockquote', () => {
  const markdownText = `This is
  > A quote.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a multiline html block', () => {
  const markdownText = `This is a custom
  
  <custom src="property">
  contents
  </custom>

  block.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip an ordered list', () => {
  const markdownText = `This is an ordered list:
1. one
1. two
1. three

Done.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip an unordered list', () => {
  const markdownText = `This is an unordered list:
* one
* two
* three

Done.`;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a single line html block', () => {
  const markdownText = 'This is a <custom src="property"/> property.';
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

test('can roundtrip a code block', () => {
  const markdownText = `\`\`\`
  this is a multiline
  code
  block.
  \`\`\``;
  const value = fromMarkdown.convert(markdownText);
  const markdownRound = toMarkdown.convert(value);
  // expect(markdownRound).toMatch(markdownText);
  const valueRound = fromMarkdown.convert(markdownRound);
  expect(valueRound.toJSON()).toMatchSnapshot();
});

