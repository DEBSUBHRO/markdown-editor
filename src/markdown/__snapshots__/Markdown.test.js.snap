// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can roundtrip a blockquote 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "A quote.",
          },
        ],
        "object": "block",
        "type": "block_quote",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip a code block 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "  this is a multiline
  code
  block.
",
              },
            ],
            "object": "block",
            "type": "paragraph",
          },
        ],
        "object": "block",
        "type": "code_block",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip a link 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is ",
          },
          Object {
            "data": Object {
              "href": "http://clause.io",
            },
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "a link",
              },
            ],
            "object": "inline",
            "type": "link",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": ".",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip a multiline html block 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is a custom",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [
                  Object {
                    "data": Object {},
                    "object": "mark",
                    "type": "html",
                  },
                ],
                "object": "text",
                "text": "  <custom src=\\"property\\">
  contents
  </custom>",
              },
            ],
            "object": "block",
            "type": "paragraph",
          },
        ],
        "object": "block",
        "type": "html_block",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "block.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip a single line html block 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is a ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "html",
              },
            ],
            "object": "text",
            "text": "<custom src=\\"property\\"/>",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": " property.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip a thematic break 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [],
        "object": "block",
        "type": "horizontal_rule",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "A footer.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip an ordered list 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is an ordered list:",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "one",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "two",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "three",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
        ],
        "object": "block",
        "type": "ol_list",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Done.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip an unordered list 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is an unordered list:",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "one",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "two",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "marks": Array [],
                "object": "text",
                "text": "three",
              },
            ],
            "object": "block",
            "type": "list_item",
          },
        ],
        "object": "block",
        "type": "ul_list",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Done.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip basic text 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is some text.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip bold italic text 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "bold",
              },
              Object {
                "data": Object {},
                "object": "mark",
                "type": "italic",
              },
            ],
            "object": "text",
            "text": "some",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": " text.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip bold text 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "bold",
              },
            ],
            "object": "text",
            "text": "some",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": " text.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h1 to slate 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading One",
          },
        ],
        "object": "block",
        "type": "heading_one",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h2 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading Two",
          },
        ],
        "object": "block",
        "type": "heading_two",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h3 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading Three",
          },
        ],
        "object": "block",
        "type": "heading_three",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h4 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading Four",
          },
        ],
        "object": "block",
        "type": "heading_four",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h5 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading Five",
          },
        ],
        "object": "block",
        "type": "heading_five",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip h6 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Heading Six",
          },
        ],
        "object": "block",
        "type": "heading_six",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip inline code 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "code",
              },
            ],
            "object": "text",
            "text": "inline code",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": ".",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip italic text 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is ",
          },
          Object {
            "marks": Array [
              Object {
                "data": Object {},
                "object": "mark",
                "type": "italic",
              },
            ],
            "object": "text",
            "text": "some",
          },
          Object {
            "marks": Array [],
            "object": "text",
            "text": " text.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`can roundtrip paragraphs 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is first paragraph.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "This is second paragraph.",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;
